File: 백준/Silver/11057. 오르막 수/오르막 수.py
n = int(input())
 
dp = [[1] * 10 for _ in range(n)]
 
for i in range(1, n):
  for j in range(1, 10):
    dp[i][j] = (dp[i][j-1] + dp[i-1][j])%10007
 
print(sum(dp[n-1])%10007)

File: 백준/Silver/11057. 오르막 수/오르막 수.js
let fs = require('fs');
let inputs = fs.readFileSync('/dev/stdin').toString().trim();
const N = Number(inputs);

const dp = Array.from({length: N}, () => Array.from({length: 10}, () => 0));

for (let i = 0; i < 10; i++) {
    dp[0][i] = 1;
}

for (let i = 1; i < N; i++) {
    for (let j = 0; j < 10; j++) {
        for (let k = 0; k <= j; k++) {
            dp[i][j] += (dp[i - 1][k] % 10007);
        }
    }
}

let sum = 0;
for (const ele of dp[N - 1]) {
    sum += ele;
}

console.log(sum % 10007);